src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class CachePoolClearerPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        foreach ($container->findTaggedServiceIds('cache.pool') as $id => $tags) {
+            foreach ($tags as $attr) {
+            }
+            if (isset($attr['clearer'])) {
+                $clearer = $container->getDefinition($attr['clearer']);
+                $clearer->addMethodCall('addPool', array(new Reference($id)));
```

Comment:
I just wanted to say that I see no reason to prefer setter injection over constructor. Same level of coupling for me, better performance of constructor injection, cleaner API, less side-effects, etc. 

I see that the class has no constructor, but it's not an issue to add it.

Even if somebody added his own implementation, he'd look into this source code and find out that he needs to add `addPool` method or adopt constructor signature, i.e. same kind of changes. How setter is looser coupling?